mlir/test/Dialect/Linalg/vectorization.mlir

```diff
@@ -501,6 +501,67 @@ module attributes {transform.with_named_sequence} {
 
 // -----
 
+func.func @test_vectorize_dynamic_pack(%arg0: tensor<?x?xf32>, %arg1: tensor<4x1x16x2xf32>) -> tensor<4x1x16x2xf32> {
+  %pack = tensor.pack %arg0 inner_dims_pos = [1, 0] inner_tiles = [16, 2] into %arg1 : tensor<?x?xf32> -> tensor<4x1x16x2xf32>
+  return %pack : tensor<4x1x16x2xf32>
+}
+module attributes {transform.with_named_sequence} {
+  transform.named_sequence @__transform_main(%arg0: !transform.any_op {transform.readonly}) {
+    %0 = transform.structured.match ops{["tensor.pack"]} in %arg0 : (!transform.any_op) -> !transform.any_op
+    transform.structured.vectorize %0 vector_sizes [8, 16] : !transform.any_op
```

Comment:
I think it is better to keep consistent with the TilingInterface implementation. When we provide sizes to the op, they are applied on destination outer dimensions. In this context, the vector_sizes should be `[4, 1]`. What do you think?

Reply:
This will also be aligned with how we set lowering_config in IREE. We can look at lowering_config when inferring input vector sizes gets tricky.